server/routes/upload.js

```diff
@@ -0,0 +1,47 @@
+const Profile = require("../models/Profile");
+const express = require('express');
+const AWS = require('aws-sdk');
+const multer = require('multer');
+const   multerS3 = require('multer-s3');
+const router = express.Router()
+require('dotenv').config();
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.AWS_ID,
+  secretAccessKey: process.env.AWS_KEY
+});
+
+var upload = multer({
+  storage: multerS3({
+    s3: s3,
+    bucket: process.env.BUCKET_NAME,
+    key: function (req, file, cb) {
+      cb(null, file.originalname); 
+    }
+  })
+});
+
+router.post('/uploadImage', upload.single('image'), function(req, res, next) {
```

Comment:
```suggestion
router.post('/uploadImage', upload.single('image'), function(req, res, next) {
```
```suggestion
router.post('/upload-image', upload.single('image'), function(req, res, next) {
```
We use hyphenated case for route names in JS :)

Reply:
Let's also apply the `protect` middleware to this route so that only authorized users can upload profile photos

Reply:
not sure where i add the protect middleware...

Reply:
Do i just make a single middleware function?

const middleware = function(req, res, next){ 
   protect
   upload.single('image')
} 

and then put that in router.post('uploadImage', middleware, function(req, res, next){}

Reply:
@rajivtitus